test/e2e/e2e_suite_test.go

```diff
@@ -80,6 +86,71 @@ var (
 	useCIArtifacts bool
 )
 
+type (
+	AzureClusterProxy struct {
+		framework.ClusterProxy
+	}
+)
+
+func NewAzureClusterProxy(name string, kubeconfigPath string, scheme *runtime.Scheme, options ...framework.Option) *AzureClusterProxy {
+	return &AzureClusterProxy{
+		ClusterProxy: framework.NewClusterProxy(name, kubeconfigPath, scheme, options...),
+	}
+}
+
+func (acp *AzureClusterProxy) CollectWorkloadClusterLogs(ctx context.Context, namespace, name, outputPath string) {
+	const (
+		kubesystem = "kube-system"
+	)
+
+	Byf("Dumping workload cluster %s/%s logs", namespace, name)
+	acp.ClusterProxy.CollectWorkloadClusterLogs(ctx, namespace, name, outputPath)
+
+	Byf("Dumping workload cluster %s/%s kube-system pod logs", namespace, name)
+	aboveMachinesPath := strings.Replace(outputPath, "/machines", "", 1)
+	workload := acp.GetWorkloadCluster(ctx, namespace, name)
+	pods := &corev1.PodList{}
+	Expect(workload.GetClient().List(ctx, pods, client.InNamespace(kubesystem))).To(Succeed())
```

Comment:
none of this is CAPZ specific and it would useful to have kube-system pod logs for all CAPI clusters, would it make sense to make this change directly in the framework's log collector instead?

Reply:
It would, but I want it today, not in the next CAPI release :).

I will gladly open a similar PR in CAPI.